From 14b9436cc93716177c5c6a4e0e392ca68f53b2d0 Mon Sep 17 00:00:00 2001
From: Brandon Williams <turbine.bmw@gmail.com>
Date: Sun, 12 Jul 2026 19:19:54 +0000
Subject: [PATCH 1/3] libcamera: camera_sensor_properties: Add IMX681

Describe the 1.0 um unit cell and the two-frame exposure, gain and
blanking control delays for the Sony IMX681.

Signed-off-by: Brandon Williams <turbine.bmw@gmail.com>
---
 src/libcamera/sensor/camera_sensor_properties.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/libcamera/sensor/camera_sensor_properties.cpp b/src/libcamera/sensor/camera_sensor_properties.cpp
index 28043150e..cf40f26f8 100644
--- a/src/libcamera/sensor/camera_sensor_properties.cpp
+++ b/src/libcamera/sensor/camera_sensor_properties.cpp
@@ -319,6 +319,16 @@ const CameraSensorProperties *CameraSensorProperties::get(const std::string &sen
 				.hblankDelay = 2
 			},
 		} },
+		{ "imx681", {
+			.unitCellSize = { 1000, 1000 },
+			.testPatternModes = {},
+			.sensorDelays = {
+				.exposureDelay = 2,
+				.gainDelay = 2,
+				.vblankDelay = 2,
+				.hblankDelay = 2
+			},
+		} },
 		{ "imx708", {
 			.unitCellSize = { 1400, 1400 },
 			.testPatternModes = {
-- 
2.55.0

